### ISSUE TITLE
botti doesn't build anymore with --with-perl=yes

### ISSUE THREAD

**Issue by ailin-nemui** [irssi/irssi#1374]
1.3.2

reported by tomaw

also reported by Unit193 in https://github.com/irssi/irssi/issues/1345

**Comment by yarda** [comment]
irssi-1.4.1 in Fedora Rawhide, the same problem

**Comment by yarda** [comment]
We used the following hack in Fedora downstream:
https://src.fedoraproject.org/rpms/irssi/blob/rawhide/f/irssi-1.4.1-botti-perl-link-fix.patch

Meson build untested.

**Comment by yarda** [comment]
The problem also occurs with: --with-perl=module

**Comment by currificado** [comment]
The patch worked for me, although I had to apply it manually.

**Comment by diddledani** [comment]
Drive-by comment from the snapcrafters community. @ailin-nemui you mention you are wanting feedback from users, but haven't stated what feedback you require. Perhaps you could update your first post to indicate what you want to learn from users so that they know to add their thoughts here? As it stands, there is no request for information or feedback apart from your comment further down the thread in response to a patch - users are unlikely to see that, and even if they did they won't know what feedback or information you need.

**Comment by currificado** [comment]
> @currificado are you using botti?

Yes, I am. Sorry that I forgot to reply your question. I have two bots and one of them consists of just irssi modules that are run with botti.

**Comment by RJ-Frosty** [comment]
After running `meson Build -Dwith-perl=yes -Dwithout-textui=yes` 
`ninja -C` doesn't work
`[233/234] Linking target tests/irc/flood/test-796
ninja: build stopped: subcommand failed.`
Not using botti, I am planing to use irssix.

**Comment by esaaprillia** [comment]
@yarda 

Thank you for your patch. the same problem I encountered. now I have successfully built irssi in OpenWrt with perl

```
root@OpenWrt:/usr/bin# ldd irssi
        /lib/ld-musl-aarch64.so.1 (0x7f8e39e000)
        libperl.so => /usr/lib/perl5/5.28/CORE/libperl.so (0x7f8e083000)
        libgmodule-2.0.so.0 => /usr/lib/libgmodule-2.0.so.0 (0x7f8e06f000)
        libglib-2.0.so.0 => /usr/lib/libglib-2.0.so.0 (0x7f8df3e000)
        libssl.so.1.1 => /usr/lib/libssl.so.1.1 (0x7f8deab000)
        libcrypto.so.1.1 => /usr/lib/libcrypto.so.1.1 (0x7f8dc40000)
        libncursesw.so.6 => /usr/lib/libncursesw.so.6 (0x7f8dbdb000)
        libgcc_s.so.1 => /lib/libgcc_s.so.1 (0x7f8dbb8000)
        libc.so => /lib/ld-musl-aarch64.so.1 (0x7f8e39e000)
root@OpenWrt:/usr/bin#
```
